test_programs/compile_failure/negate_unsigned/src/main.nr

```diff
@@ -0,0 +1,6 @@
+use dep::std;
+
+fn main() {
+    let var = -1 as u8;
```

Comment:
I think this is not catching the intended error. On master this fails for me with a different error message:
```
error: Assertion failed: 'attempt to subtract with overflow'
```

It looks like a negative Field value being cast to a `u8` after the negation. Maybe we need to do `let one: u8 = 1; let var = -one;`

Reply:
Alternatively `let var: u8 = -1;` should also work. Although with the negative integer literals PR this would soon become a negative literal instead of a unary operation.

Reply:
On this branch we will get `u8 cannot be used in a unary operation`. Perhaps we could instead expand `TypeCheckError::InvalidUnaryOp` to include the specific unary operation.

Reply:
@vezenovm, yes that's a good point, I did it.